app/services/user_alerts/alert_user_about_new_device.rb

```diff
@@ -3,19 +3,37 @@
 module UserAlerts
   class AlertUserAboutNewDevice
     def self.call(user, device, disavowal_token)
-      device_decorator = DeviceDecorator.new(device)
-      login_location = device_decorator.last_sign_in_location_and_ip
-      device_name = device_decorator.nice_name
+      if IdentityConfig.store.feature_new_device_alert_aggregation_enabled
+        user.sign_in_new_device_at ||= Time.zone.now
+        user.save
+      else
+        device_decorator = DeviceDecorator.new(device)
+        login_location = device_decorator.last_sign_in_location_and_ip
+        device_name = device_decorator.nice_name
 
-      user.confirmed_email_addresses.each do |email_address|
-        UserMailer.with(user: user, email_address: email_address).new_device_sign_in(
-          date: device.last_used_at.in_time_zone('Eastern Time (US & Canada)').
-            strftime('%B %-d, %Y %H:%M Eastern Time'),
-          location: login_location,
-          device_name: device_name,
-          disavowal_token: disavowal_token,
-        ).deliver_now_or_later
+        user.confirmed_email_addresses.each do |email_address|
+          UserMailer.with(user: user, email_address: email_address).new_device_sign_in(
+            date: device.last_used_at.in_time_zone('Eastern Time (US & Canada)').
+              strftime('%B %-d, %Y %H:%M Eastern Time'),
+            location: login_location,
+            device_name: device_name,
+            disavowal_token: disavowal_token,
+          ).deliver_now_or_later
+        end
```

Comment:
> > We probably want the feature flag to prevent the _current_ "new device sign-in" message being sent when enabled. We could consider that as part of the follow-up "wiring up" work.
> 
> I thought it was preventing it from being sent with this conditional. Or would something more explicit need to happen? (in a follow on ticket?)

Ah you're right, I overlooked that we were altering the logic for the existing notification ehre.